### DISCUSSION TITLE
What happens if I set base url from import_css ?

### DISCUSSION THREAD

**Discussion by octalflow** [litehtml/litehtml#372] [Q&A] [answered]
Hi.
import_css(litehtml::string& text, const litehtml::string& url, litehtml::string& baseurl)
{
text = read_file( someFile );
baseurl = "a/b/c";
}

I set the baseurl "a/b/c" .

but all the function call after import_css, I see baseurl  is never used including get_img_size.

Am I miss something ?

**Comment by tordex** [comment, marked as answer]
the ```baseurl``` you set in the ```import_css``` is applied to the URLs  in this imported CSS only. For example for background images.

Generally the rules for baseurl:
1. Overwrite the function ```set_base_url``` to save base url from ```<base>``` tag
2. Calculate default baseurl based on the url of the opened document. litehtml doesn't know about document url, but ```document_container``` implementation should know it.
3. In the "images function" use ```baseurl``` argument if it is not empty. If ```baseurl``` is empty, use the calculated default baseurl.
